server/websocket/src/server.rs

```diff
@@ -137,19 +158,30 @@ where
                             let service = GraphQlConnection::new(
                                 &logger,
                                 subgraphs.clone(),
-                                subgraph,
+                                subgraph.clone(),
                                 ws_stream,
                                 graphql_runner.clone(),
                             );
-                            tokio::spawn(service.into_future()).into_future()
+
+                            let cancel_subgraph = subgraph.clone();
+                            let (service, guard) = service.into_future().cancelable(move || {
+                                debug!(
+                                    logger,
+                                    "Canceling subscriptions for subgraph `{}`", cancel_subgraph
```

Comment:
We've been _fairly_ consistent in using structured logging, so it would be nice to use that here as well:
```
debug!(
    logger,
    "Canceling subscriptions for subgraph";
    "subgraph_name" => ...
    "subgraph_id" => ...,
);
```